### PR TITLE
ci(mergify): use mergify to request reviewers

### LINKED ISSUE #1654: [Bug] `/request_review` PR workflow runs fail due to apparent permissions issue
Didn't work:

https://github.com/cryostatio/cryostat/actions/runs/6087853970/job/16517292637

```
 Adding label 'review-requested' to the PR
https://github.com/cryostatio/cryostat/pull/1636
Requesting review from @cryostatio/reviewers
error fetching organization teams: GraphQL: Resource not accessible by integration (organization.teams)
Error: Process completed with exit code 1.
```

I checked the organization settings and it looks like the reviewers team is publicly readable, so I think this workflow might need to run with some elevated token permissions?

_Originally posted by @andrewazores in https://github.com/cryostatio/cryostat/issues/1638#issuecomment-1707029937_

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes #1654 

## Description of the change:

Use Mergify to request reviewers since it is granted sufficient permissions: https://docs.mergify.com/getting-started/#required-permissions

Workflow is used to add label and trigger Mergify actions.

## Motivation for the change:

See #1654

### CODE REVIEW

**Comment by tthvo** [comment]
This PR setups Mergify to:
- If the `review-requested` label is applied, it will request reviews from `cryostatio/reviewers`.
- If there are already some reviews (i.e. comment, requests or approval), the label will be removed. This means running the command `/request_review` 2nd time after reviews are in, the label will be removed immediately.

I suppose the command `/request_review` is meant to be called once when the PR is ready?

**Comment by tthvo** [comment]
Also, `safe-to-test` label is added using `gh pr edit 1657 --add-label "safe-to-test" --repo cryostatio/cryostat`. So, we can just use the `--repo` option here and remove `action/checkout`.

**Comment by aali309** [comment]
> This PR setups Mergify to:
> 
> * If the `review-requested` label is applied, it will request reviews from `cryostatio/reviewers`.
> * If there are already some reviews (i.e. comment, requests or approval), the label will be removed. This means running the command `/request_review` 2nd time after reviews are in, the label will be removed immediately.
> 
> I suppose the command `/request_review` is meant to be called once when the PR is ready?

From what I understood, if you comment again  `/request_review` it should ask for review again and the label should be added again too.

**Comment by andrewazores** [comment]
Yep I agree with that, I think the workflow will just be that the `/request_review` can be commented at any time to re-add the label and ensure that the reviewers team is assigned. The label will be removed whenever any reviewer does review the PR in any form. This back-and-forth repeats until the PR is merged.

File: .mergify.yml [hunk 1]
```diff
@@ -39,3 +39,24 @@ pull_request_rules:
...
             required frontend work by opening and merging a PR against
             [cryostat-web](https://github.com/cryostatio/cryostat-web).
+
+  - name: request reviews from the reviewers team
+    conditions:
+      - label=review-requested
+    actions:
+      request_reviews:
+        teams:
+          - "@reviewers"
```
Comment (aali309) [inline .mergify.yml:49]: `@reviewers` did you mean `cryostatio/reviewers` ?

Comment (aali309) [reply .mergify.yml:49]: /request_review

Comment (andrewazores) [reply .mergify.yml:49]: Good catch: https://docs.mergify.com/actions/request_reviews/#flexible-reviewers-assignment

**Comment by andrewazores** [comment]
I don't think the new setup will trigger until after this is merged, so it can't be tested on this PR itself unfortunately.

**Comment by tthvo** [comment]
Thanks for clarifying! I updated such that the label will be added for subsequent commands. Tho, the reviewers will basically no-op since they are already requested.

**Comment by aali309** [comment]
> Thanks for clarifying! I updated such that the label will be added for subsequent commands. Tho, the reviewers will basically no-op since they are already requested.

Looks good to me.

File: .mergify.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ pull_request_rules:
 
   - name: auto label PRs from reviewers
     conditions:
-      - author=@reviewers
+      - author=@cryostatio/reviewers
```
Comment (andrewazores) [inline .mergify.yml:15]: I don't think the Mergify docs are particularly clear or consistent about this:

https://docs.mergify.com/conditions/#attributes

https://docs.mergify.com/actions/request_reviews/#flexible-reviewers-assignment

It seems like in the `actions: foo: teams:` attribute you must specify the whole `@org/teamname`, but in `conditions:` it is just `author=@teamname`

Comment (tthvo) [reply .mergify.yml]: Ahh right! Updated now :))

**Review by andrewazores — APPROVED** [review]

**Pull request merged**